karbon/karbon_cluster_registration_structs.go

```diff
@@ -26,20 +26,43 @@ type K8sClusterRegistrationDeleteResponse struct {
 	TaskUUID *string `json:"task_uuid"`
 }
 
+// K8sClusterInfo K8s cluster info
+type K8sClusterInfo struct {
+	// k8s distribution
+	// Enum: [Openshift NKE CAPX]
+	K8sDistribution string `json:"k8s_distribution,omitempty"`
+
+	// k8s version
+	K8sVersion string `json:"k8s_version,omitempty"`
+}
+
+// K8sClusterAddonInfo K8s cluster addon info
+type K8sClusterAddonInfo struct {
+	// addon name
+	AddonName string `json:"addon_name,omitempty"`
+
+	// addon version
+	AddonVersion string `json:"addon_version,omitempty"`
+}
+
+// K8sClusterRegistration K8s cluster registration details.
 type K8sClusterRegistration struct {
+	// addons info
+	AddonsInfo []*K8sClusterAddonInfo `json:"addons_info,omitempty"`
+
 	// Categories for this k8s cluster. This allows setting up multiple values from a single key.
 	CategoriesMapping map[string]string `json:"categories_mapping,omitempty"`
 
-	// identity
-	Identity *K8sIdentity `json:"identity,omitempty"`
```

Comment:
Will this no longer be part of the response? If so, do existing mocks need to be regenerated?

Reply:
yes, ideally.

Reply:
it was never passed in request so it never made it to yamls so no change required